lightstep/provider.go

```diff
@@ -30,7 +30,14 @@ func Provider() *schema.Provider {
 				Optional:     true,
 				DefaultFunc:  schema.EnvDefaultFunc("LIGHTSTEP_ENV", "public"),
 				ValidateFunc: validation.StringMatch(regexp.MustCompile(`^(staging|meta|public)$`), "Must be one of: staging, meta, public"),
-				Description:  "The name of the Lightstep environment, must be one of: staging, meta, public.",
+				Description:  "(deprecated) The name of the Lightstep environment, must be one of: staging, meta, public. Deprecated in favor of `api_url`",
+				Deprecated:   "This field is deprecated and will be removed in a future release. Please use the `api_url` field instead.",
+			},
+			"api_url": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("LIGHTSTEP_API_BASE_URL", ""),
```

Comment:
Thoughts on whether this should be called `api_base_url` or if the environment variable should be named `LIGHTSTEP_API_URL` to be consistent?

Reply:
It probably should, but I wanted to reuse the current environment variable to avoid breaking changes

Reply:
Ah okay. I don't think that's necessarily a problem if it's existed for a while. The only question I'd have is: Are we changing the meaning of the original environment variable? I don't think we are, but figured I'd ask.

Reply:
Perhaps we could have it support both `LIGHTSTEP_API_URL` and `LIGHTSTEP_API_BASE_URL`? If one isn't set, then look for the other. People coming in fresh would then not have to care about the old way.

Reply:
good point, looks like there's a helper function for this. updated.